### DISCUSSION TITLE
Generic 2D and 3D Cards

### DISCUSSION THREAD

**Discussion by LeoKids** [86Box/86Box#4144] [Ideas]
I would want an generic card that supports 2D and 3D acceleration.

Why:
Because the GPU selection only goes to voodoo 3 which is more outdated,And RIVA TNT/Geforce and ATI Rage/Radeon can't be added cause of lack of documentation,To fill for the lack of modern-ish cards,we need a generic card that supports those things.

What would 2d card have:
1. 2d acceleration
2. DDraw 1-7 support,typically unneccesary as we may already have an DDraw cards on there
3. All the things 2D Cards might have
4. Support for DOS compability and VESA 1.0-3.0

What would 3D card have:
1. 3D acceleration
2. Direct3D acceleration (up to DX7 in hardware,supports up to dx9 in software)
3. Shader Model 1.x,2.0 and later (2.0 and later required for Aero)
4. Native DX8.DX9,DX10,DX11,DX12 support (Modern DirectX like DX10,DX11 and DX12 is unnecessary,due to the fact 86box cannot run Windows 8 and later but can be done)
5. Hardware T&L (required for games like The Sims 2 to run fast)

This might be a great idea and it should be implemented.

**Comment by Torinde** [comment]
I think the above can be accomplished by providing support for some of the [VirtualBox/QEMU graphics devices](https://github.com/86Box/86Box/discussions/4573) (and maybe extending them, if not everything listed above is supported currently)

**Comment by pankoza2-pl** [comment]
For 2D, Bochs SVGA is now added (for quite some time now)
